packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
+    logging.info('VIP_TEST START {}'.format(teststr))
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not samehost:
+        host2 = agents[1]
+    logging.info('host1 is is: {}'.format(host1))
+    logging.info('host2 is is: {}'.format(host2))
+    origin_app, app_uuid = get_test_app_in_docker()
+    origin_app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': vip}
+    origin_app['container']['docker']['network'] = vipnet
+    if vipnet == 'USER':
+        origin_app['ipAddress'] = {'networkName': 'dcos'}
+    if vipnet == 'HOST':
+        origin_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        origin_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
+        origin_app['cmd'] += '$PORT0'
+        origin_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['labels'] = {}
+        origin_app['portDefinitions'] = [{'labels': {'VIP_0': vip}}]
+
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+
+    proxy_app = get_test_app_in_docker()[0]
+    proxy_app['container']['docker']['network'] = proxynet
+    proxy_app['constraints'] = [['hostname', 'CLUSTER', host2]]
+    if proxynet == 'USER':
+        proxy_app['ipAddress'] = {'networkName': 'dcos'}
+    if proxynet == 'HOST':
+        proxy_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '
+        proxy_app['cmd'] += '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py '
+        proxy_app['cmd'] += '$PORT0'
+        proxy_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        proxy_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+
+    returned_uuid = None
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app))
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)()
+        logging.info('returned_uuid is: {}'.format(returned_uuid))
+    assert(returned_uuid is not None)
+    assert(returned_uuid['test_uuid'] == app_uuid)
+    logging.info('VIP_TEST PASSED {}'.format(teststr))
+    return app_uuid
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_vip(dcos_api_session):
+    """Test every permutation of VIP
+    """
+    num = 0
+    addrs = [['1.1.1.{}:{}', '1.1.1.{}:{}'],
+             ['/namedvip{}:{}', 'namedvip{}.marathon.l4lb.thisdcos.directory:{}']]
```

Comment:
you mean the parameter fixtures?  There is no way to run all those tests in parallel.  If run sequentially they take 40 minutes :).